docs/dev/table/tuning/streaming_aggregation_optimization.md

```diff
@@ -0,0 +1,271 @@
+
+The following examples show how to enable the split distinct aggregation optimization.
+
+<div class="codetabs" markdown="1">
+<div data-lang="java" markdown="1">
+{% highlight java %}
+// instantiate table environment
+TableEnvironment tEnv = ...
+
+tEnv.getConfig()        // access high-level configuration
+  .getConfiguration()   // set low-level key-value options
+  .setString("table.optimizer.distinct-agg.split.enabled", "true");  // enable distinct agg split
+{% endhighlight %}
+</div>
+
+<div data-lang="scala" markdown="1">
+{% highlight scala %}
+// instantiate table environment
+val tEnv: TableEnvironment = ...
+
+tEnv.getConfig         // access high-level configuration
+  .getConfiguration    // set low-level key-value options
+  .setString("table.optimizer.distinct-agg.split.enabled", "true")  // enable distinct agg split
+{% endhighlight %}
+</div>
+
+<div data-lang="python" markdown="1">
+{% highlight python %}
+# instantiate table environment
+t_env = ...
+
+t_env.get_config()        # access high-level configuration
+  .get_configuration()    # set low-level key-value options
+  .set_string("table.optimizer.distinct-agg.split.enabled", "true"); # enable distinct agg split
+{% endhighlight %}
+</div>
+</div>
+
+## Use FILTER Modifier on Distinct Aggregates
+
+In some cases, user may need to calculate the number of UV (unique visitor) from different dimensions, e.g. UV from Android, UV from iPhone, UV from Web and the total UV.
+Many users will choose `CASE WHEN` to support this, for example:
+
+{% highlight sql %}
+SELECT
+ day,
+ COUNT(DISTINCT user_id) AS total_uv,
+ COUNT(DISTINCT CASE WHEN flag IN ('android', 'iphone') THEN user_id ELSE NULL END) AS app_uv,
+ COUNT(DISTINCT CASE WHEN flag IN ('wap', 'other') THEN user_id ELSE NULL END) AS web_uv
+FROM T
+GROUP BY day
+{% endhighlight %}
+
+However, it is recommended to use `FILTER` syntax instead of CASE WHEN in this case. Because `FILTER` is more compliant with the SQL standard and will get much more performance improvement.
+`FILTER` is a modifier used on an aggregate function to limit the values used in an aggregation. Replace the above example with `FILTER` modifier as following:
+
+{% highlight sql %}
+SELECT
+ day,
+ COUNT(DISTINCT user_id) AS total_uv,
+ COUNT(DISTINCT user_id) FILTER (WHERE flag IN ('android', 'iphone')) AS app_uv,
+ COUNT(DISTINCT user_id) FILTER (WHERE flag IN ('wap', 'other')) AS web_uv
+FROM T
+GROUP BY day
+{% endhighlight %}
+
+Flink SQL optimizer can recognize the different filter arguments on the same distinct key. For example, in the above example, all the three COUNT DISTINCT are on `user_id` column.
+Then Flink can use just one share state instance instead of three state instances to reduce state access and state size. In some workloads, this can get significant performance improvements.
```

Comment:
replacing “share” with "shared" here would be better?
“just one **share** state instance”->“just one **shared** state instance”